### PR TITLE
Show license expired message on Version History page

### PR DESCRIPTION
**Pull request by divolgin**
#### What this PR does / why we need it:

In Helm managed mode, when checking for updates, the API will return the "license expired" message.

Screen Shot 2022-09-28 at 9 55 07 AM

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
* Fixes an issue where checking for updates from the [version history](/enterprise/updating-apps#update-an-application-in-the-admin-console) page would not show the "License is expired" error when the check fails due to an expired license in [Helm managed mode (Alpha)](/vendor/helm-install).
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: web/src/components/apps/AppVersionHistory.jsx [hunk 1]
```diff
@@ -1002,7 +1002,7 @@ class AppVersionHistory extends Component {
       });
 
       const res = await fetch(
-        `${process.env.API_ENDPOINT}/app/${app?.slug}/cluster/${clusterId}/sequence/${version?.sequence}/downstreamoutput`,
+        `${process.env.API_ENDPOINT}/app/${app.slug}/cluster/${clusterId}/sequence/${version.sequence}/downstreamoutput`,
```
Comment (divolgin) [inline web/src/components/apps/AppVersionHistory.jsx:1005]: Slug and version are required here.  If they are missing, UI state is not managed right.

File: web/src/components/apps/DashboardVersionCard.jsx [hunk 2]
```diff
@@ -481,20 +483,22 @@ class DashboardVersionCard extends React.Component {
...
-                    currentVersion?.status === "failed"
-                  )
-                }
-                data-tip="View deploy logs"
-              >
-                <Icon icon="view-logs" size={22} className="clickable" />
-              </span>
-              <ReactTooltip effect="solid" className="replicated-tooltip" />
-            </div>
+            {app ? (
```
Comment (divolgin) [inline web/src/components/apps/DashboardVersionCard.jsx:486]: Can't click without app.  Was causing random failures in tests.

File: pkg/updatechecker/updatechecker.go [hunk 3]
```diff
@@ -246,6 +247,23 @@ func checkForHelmAppUpdates(opts CheckForUpdatesOpts, finishedChan chan<- error)
 
 	availableUpdateTags, err := helm.CheckForUpdates(helmApp, license, &currentVersion)
 	if err != nil {
+		expiredErr := util.ActionableError{
+			NoRetry: true,
+			Message: "License is expired",
+		}
+		isExpired := false
+
+		newLicense, _ := helm.GetChartLicenseFromSecretOrDownload(helmApp)
```
Comment (sgalsaleh) [inline pkg/updatechecker/updatechecker.go:256]: why not check for errors?

**Review by sgalsaleh — APPROVED** [review]

File: pkg/updatechecker/updatechecker.go (hunk 3 above)
Comment (divolgin) [reply pkg/updatechecker/updatechecker.go:256]: if we can't sync and local copy is expired, what else is there to do....

**Pull request merged**